codegen/generator/go/templates/src/header.go.tmpl

```diff
@@ -4,7 +4,20 @@ package {{ .Package }}
 
 import (
 	"context"
+	"reflect"
 
 	"github.com/Khan/genqlient/graphql"
 	"dagger.io/dagger/internal/querybuilder"
 )
+
+// assertNotNil panic if the given value is nil.
+// This function is used to validate that input with pointer type are not nil.
+// See https://github.com/dagger/dagger/issues/5696 for more context.
+func assertNotNil(value any) {
+	// We use reflect because just comparing value to nil is not working since
+	// the value is wrapped into a type when passed as parameter.
+	// E.g., nil become (*dagger.File)(nil).
+	if reflect.ValueOf(value).IsNil() {
+		panic("unexpected nil pointer")
+	}
+}
```

Comment:
I think it should not be part of generate. Create a new go source file would make more sense?

Reply:
I do not have a strong opinon about that, but since this function shall only be called in the `api.gen.go` file, why should it be in another file?